sentry_sdk/integrations/asyncpg.py

```diff
@@ -0,0 +1,236 @@
+        )
+
+
+T = TypeVar("T")
+
+
+def _wrap_execute(f: Callable[..., Awaitable[T]]) -> Callable[..., Awaitable[T]]:
+    async def _inner(*args, **kwargs) -> T:
+        hub = Hub.current
+        integration = hub.get_integration(AsyncPGIntegration)
+
+        # Avoid recording calls to _execute twice.
+        # Calls to Connection.execute with args also call
+        # Connection._execute, which is recorded separately
+        # args[0] = the connection object, args[1] is the query
+        if integration is None or len(args) > 2:
+            return await f(*args, **kwargs)
+
+        query = args[1]
+        with record_sql_queries(hub, None, query, None, None, executemany=False):
+            res = await f(*args, **kwargs)
+        return res
+
+    return _inner
+
+
+SubCursor = TypeVar("SubCursor", bound=BaseCursor)
+
+
+@contextlib.contextmanager
+def _record(
+    hub: Hub,
+    cursor: SubCursor | None,
+    query: str,
+    params_list: tuple | None,
+    *,
+    executemany: bool = False
+):
+    integration = hub.get_integration(AsyncPGIntegration)
+    if not integration._record_params:
+        params_list = None
+
+    param_style = "pyformat" if params_list else None
+
+    with record_sql_queries(
+        hub,
+        cursor,
+        query,
+        params_list,
+        param_style,
+        executemany=executemany,
+        record_cursor_repr=cursor is not None,
+    ) as span:
+        yield span
+
+
+def _wrap_connection_method(
+    f: Callable[..., Awaitable[T]], *, executemany=False
+) -> Callable[..., Awaitable[T]]:
+    async def _inner(*args: Any, **kwargs: Any) -> T:
+        hub = Hub.current
+        integration = hub.get_integration(AsyncPGIntegration)
+
+        if integration is None:
+            return await f(*args, **kwargs)
+
+        query = args[1]
+        params_list = args[2] if len(args) > 2 else None
+        with _record(hub, None, query, params_list, executemany=executemany) as span:
+            _set_db_data(span, args[0])
+            res = await f(*args, **kwargs)
+        return res
+
+    return _inner
+
+
+def _wrap_basecursor_exec(
+    f: Callable[..., Awaitable[T]]
+) -> Callable[..., Awaitable[T]]:
+    async def _exec(self: BaseCursor, n, timeout) -> T:
+        hub = Hub.current
+        integration = hub.get_integration(AsyncPGIntegration)
+
+        if integration is None:
+            return await f(self, n, timeout)
+
+        params_list = self._args[1] if len(self._args) > 1 else None
+
+        executemany = n > 1
+
+        with _record(
+            hub,
+            self,
+            self._query,
+            params_list,
+            executemany=executemany,
+        ):
+            res = await f(self, n, timeout)
+        return res
+
+    return _exec
+
+
+def _wrap_cursoriterator_anext(
+    f: Callable[..., Awaitable[T]]
+) -> Callable[..., Awaitable[T]]:
+    async def __await__(self: CursorIterator) -> T:  # noqa: N807
```

Comment:
We will not instrument this. Because if someone does a query returning one million rows, then one million spans will be created. That is just too much. (And in the UI it looks like the same query was run one million times. Which then would trigger Sentry's n+1 query detector, what we also do not want.)

Could you please remove the instrumentation of `__await__` of the CursorIterators anext?

Reply:
Removed it :slightly_smiling_face: 

I was thinking of maybe leaving it in and cutting it after the 10th iteration, but since the `CursorIterator` uses slots, it would be too much of a mess to fiddle the counting in.

I could, however, see this as a future improvement.